### DISCUSSION TITLE
A lot of CPU idle time during stress testing

### DISCUSSION THREAD

**Discussion by AKA-bingo** [fastapi/fastapi#11080] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import Depends, FastAPI, HTTPException
from starlette.responses import PlainTextResponse

#gunicorn --workers 1 --bind 0.0.0.0:8080 --worker-class uvicorn.workers.UvicornWorker fastapi_server.server:app
app = FastAPI()

class GetPageParams:
    def __init__(self, title: str) -> None:
        self.title = title

@app.get("/{title}")
async def get_page(params: GetPageParams=Depends()):
    if params.title:
        return PlainTextResponse(params.title)
    raise HTTPException(status_code=404, detail="Title not found")

```

### Description
I found from the flame graph that the CPU had a lot of idle wait time during the stress test
![image](https://github.com/tiangolo/fastapi/assets/7263336/35315fd8-dfb1-45d2-b0c7-de367f270fbf)
so the question is:

- What causes CPU idle when stress testing? 
- Even if blocked by some IO operations, if I can use more coroutine to accpect the requests from client?

### Operating System

Linux

### FastAPI Version

0.109.0

### Pydantic Version

2.5.3

### Python Version

Python 3.11.7

### Additional Context

App Run Command

```
gunicorn --workers 1 --bind 0.0.0.0:8000 --worker-class uvicorn.workers.UvicornWorker my_app:app
```

**Comment by Kludex** [comment]
How can I reproduce that?

**Reply by AKA-bingo** [reply]
Sorry for the Late Reply.
I just figure out the root cause is I use the `Depends()` for the request params class.
The code in question descritpion was update and you can try to reproduce in you local env. thanks~

Stree Test Command:
```
hey -c 10 -z 10m 'http://127.0.0.1:8080/{{title}}'
```

Py-spy Command:
```
py-spy record -f flamegraph -o profile.svg -i -p {{pid}}
```

**Comment by AKA-bingo** [comment]
I also find another situation that causes this problem.
I need to print the response body when the request finish. so I follow this answer to build my middleware: 
https://stackoverflow.com/a/71883126

Is there any other way to get the response body without high performance lost?
